src/programQueueProcessor/eventProcessor.ts

```diff
@@ -60,14 +78,90 @@ export default (configs: {
       };
       let newResolvedIndex: ResolvedIndex | null = null;
       await withRetry(async (retry, attemptIndex) => {
-        newResolvedIndex = await rollCallClient.createNewResolvableIndex(
-          programId.toLowerCase(),
-          true
-        );
-        logger.info(`obtained new index name: ${newResolvedIndex.indexName}`);
-        try {
-          await initIndexMapping(newResolvedIndex.indexName, esClient);
+        // check if existing latest index settings matches default settings
+        const { body } = await esClient.cat.aliases({
+          name: ROLLCALL_ALIAS_NAME,
+        });
+
+        const indices = JSON.stringify(body);
+        const regex = new RegExp(generateIndexName(programId) + "re_[0-9]+");
+        const found = indices.match(regex);
+
+        if (found) {
+          if (found.length !== 1) {
+            throw new Error(
+              `Multiple indices found for program ${programId}, unable to determin latest index name.`
+            );
+          }
+          const existingIndexName = found[0];
+          const response = await esClient.indices.getSettings({
```

Comment:
maybe give this `response` guy a type, there's a lot of `any` floating around this area...
I think something like this would work, but please double check with the actual response ;)
```
          ApiResponse<{
            [k: string]: {
              settings: {
                index: { number_of_shards: string; number_of_replicas: string };
              };
            };
          }>
```